src/declaration.h

```diff
@@ -534,6 +535,7 @@ class FuncDeclaration : public Declaration
                                         // scopes from having the same name
     VarDeclaration *vthis;              // 'this' parameter (member and nested)
     VarDeclaration *v_arguments;        // '_arguments' parameter
+    Ojbc_FuncDeclaration objc;
```

Comment:
Nearly ready, I'll probably make a PR tomorrow.  I'm just not really happy with the overhead of introducing a new type.